plugins/module_utils/network/vyos/config/bgp_global/bgp_global.py

```diff
@@ -392,16 +397,23 @@ def _bgp_global_list_to_dict(self, entry):
                     redis_dict.update({entry["protocol"]: entry})
                 proc["redistribute"] = redis_dict
 
+    def _compare_asn(self, want, have):
+        if want.get("as_number") and not have.get("as_number"):
+            self.commands.append(
+                "set protocols bgp " + "system-as" + " " + str(want.get("as_number")),
+            )
+
     def _check_af(self, neighbor):
         af_present = False
         if self._connection:
             config_lines = self._get_config(self._connection).splitlines()
             for line in config_lines:
-                if "address-family" in line:
-                    af_present = True
```

Comment:
It was necessary to check if this was relevant to the specific neighbor in order to make the output consistent with expectations in tests. Additionally, this is only used where it's looped over the changing items, so it looks ok.

Reply:
this seems to be re-added, so I am resolving this task